### PR TITLE
Add support for semver java_version and JDK 11.0.10

### PR DESCRIPTION
**Pull request by larryng**
Add support for semver `java_version`s starting with 11.0.10.

Example play:
```yaml
---
- hosts: all
  tasks:
    - name: "Install JDK 11.0.10"
      include_role:
        name: ansible-role-java
      vars:
        java_version: "11.0.10"
    - name: "Install JDK 8u131"
      include_role:
        name: ansible-role-java
      vars:
        java_version: 8
        java_subversion: 131
```

TODO:
 - [x] fix checking if already installed

### CODE REVIEW

File: tasks/set-role-variables-version-subversion.yml [hunk 1]
```diff
@@ -0,0 +1,154 @@
+---
+# file: tasks/set-role-variables-version-subversion.yml
```
Comment (larryng) [inline tasks/set-role-variables-version-subversion.yml:2]: 👀 Extracted verbatim from `set-role-variables-version.yml`.

File: tasks/set-role-variables.yml [hunk 2]
```diff
@@ -31,139 +25,30 @@
...
-    jdk_version_detail: "{{ java_version }}u{{ java_subversion }}-b17"
-  when: java_version == 8 and java_subversion == 65
-
-- name: set internal vars for 1.8.0_60
-  set_fact:
-    jdk_version_detail: "{{ java_version }}u{{ java_subversion }}-b27"
-  when: java_version == 8 and java_subversion == 60
 
-- name: set internal vars for 1.8.0_51
+- name: check if java_version is semver
```
Comment (larryng) [inline tasks/set-role-variables.yml:29]: 👀 `java_version` branching logic here

File: tasks/main.yml [hunk 3]
```diff
@@ -9,33 +9,52 @@
...
-
-- name: check if specific version of Oracle JDK is installed?
-  shell: LC_ALL="en_US.UTF-8"  {{ java_download_path }}/check-java-version.sh  "{{ jdk_version }}"
-  register: jdk_info
-  changed_when: false
-  failed_when: jdk_info.rc > 0
-
-#- debug: var=jdk_info
+# TODO: Support multiple JDK installations.
+- name: check if specific version of Oracle JDK is installed? (non-Mac)
```
Comment (larryng) [inline tasks/main.yml:25]: 👀

File: tasks/main.yml [hunk 4]
```diff
@@ -9,33 +9,52 @@
...
+      register: jdk_info
+      changed_when: false
+      failed_when: jdk_info.rc > 0
+
+    - name: set jdk_already_installed
+      set_fact:
+        jdk_already_installed: "{{ (jdk_info.stdout|from_json).found | bool }}"
+  when: ansible_os_family != "Darwin"
+
+- name: check if specific version of Oracle JDK is installed? (Mac OS X)
```
Comment (larryng) [inline tasks/main.yml:41]: 👀 Leveraging `/usr/libexec/java_home` on Mac.

File: tasks/main.yml [hunk 5]
```diff
@@ -9,33 +9,52 @@
 
 - name: precondition - java_subversion
   fail: msg="ERROR - required variable 'java_subversion' missing."
-  when: java_subversion is not defined
+  when: java_version is regex("\d+\.\d+\.\d+") and java_subversion is not defined
```
Comment (ccarpita) [inline tasks/main.yml]: I think this should be `java_version is not regex(...)`, since if it's a semver, then subversion does not need to be defined.

**Comment by ccarpita** [comment]
Other than the one potential issue, this LGTM

File: tasks/main.yml (hunk 5 above)
Comment (larryng) [reply tasks/main.yml]: oops, right

**Comment by larryng** [comment]
Updated

**Review by ccarpita — APPROVED** [review]

**Pull request merged**